### PR TITLE
CUMULUS-2895: upgrade KMS to sdk v3

### PR DESCRIPTION
**Pull request by Nnaga1**
**Summary:** Summary of changes

Addresses [CUMULUS-2895: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2895)

## Changes

- update KMS sdk code to v3

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [x] Integration tests

### CODE REVIEW

**Comment by Jkovarik** [comment]
@Nnaga1  Just a nit - please make sure to update the PR header.

File: packages/aws-client/src/KMS.ts [hunk 1]
```diff
@@ -23,12 +23,12 @@ export const createKey = (params: AWS.KMS.CreateKeyRequest = {}) =>
  * @param {string} Plaintext - the string to be encrypted
  * @returns {Promise<string>} the Base 64 encoding of the encrypted value
  */
-export const encrypt = async (KeyId: string, Plaintext: string) => {
-  const { CiphertextBlob } = await kms().encrypt({ KeyId, Plaintext }).promise();
+export const encrypt = async (KeyId: string, Plaintext: Uint8Array) => {
```
Comment (Jkovarik) [inline packages/aws-client/src/KMS.ts:26]: Yikes - this is not-insignificant change if true, right?  

At a minimum this needs a changelog entry, but we should probably also make sure our upstream code that is .js is doing the right thing here.   

e.g.

```
> textstring = new TextEncoder().encode('foo')
Uint8Array(3) [ 102, 111, 111 ]
> new TextDecoder().decode(textstring)
'foo'
``````

I may be misremembering how the varying string types work, but this needs an explination/CL entires/etc before merging.

Comment (Jkovarik) [reply packages/aws-client/src/KMS.ts:26]: If the behavior of kms.encrypt changed, we should probably keep this method signature the same and convert the string, however we need to be *very careful* this wasn't accidently mistyped and we haven't been doing the right thing here.

**Review by Jkovarik — CHANGES_REQUESTED** [review]
Noting change request from initial review comment, still reviewing.

File: packages/aws-client/src/KMS.ts (hunk 1 above)
Comment (Jkovarik) [reply packages/aws-client/src/KMS.ts:26]: Hmm, the unit doesn't indicate an issue, though, which leads me to wonder if the typing is incorrect?    What happens if we explicitly convert the string?

File: packages/aws-client/src/test-utils.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ const localStackPorts = {
   firehose: 4566,
   iam: 4566,
   kinesis: 4566,
-  kms: 4566,
+  KMS: 4566,
```
Comment (Jkovarik) [inline packages/aws-client/src/test-utils.ts:26]: What was the reason for this change? Well

File: packages/common/src/kms.ts [hunk 3]
```diff
@@ -14,13 +14,13 @@ export class KMS {
...
 
     const kms = new AWS.KMS();
-    const { CiphertextBlob } = await kms.encrypt(params).promise();
+    const CiphertextBlob = await kms.encrypt(params);
 
     if (!CiphertextBlob) {
       throw new Error('Encryption failed, undefined CiphertextBlob returned');
     }
 
-    return CiphertextBlob.toString('base64');
```
Comment (Jkovarik) [inline packages/common/src/kms.ts:23]: Why was this removed, or paerhaps more to the point why was it there in the first place?    I'm concerned with the `any` type coming back from .encrypt and this change with respect to why it was needed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Changed
 
+- *CUMULUS-2895**
```
Comment (Jkovarik) [inline CHANGELOG.md:11]: Make sure to update the CHANGELOG if needed based on comments in the PR

File: packages/aws-client/src/test-utils.ts (hunk 2 above)
Comment (etcart) [reply packages/aws-client/src/test-utils.ts:26]: this is going to be looked up by string (we don't do any caps manipulation) based on the service.serviceIdentifier or service.name parameters, which is declared in @aws-sdk/client-kms
https://github.com/nasa/cumulus/pull/3541/files does the same thing for the same reason

**Review by Jkovarik — APPROVED** [review]
The reduced scope changes look good 👍🏻  - nice work.
